app/Http/Controllers/PersonController.php

```diff
@@ -153,4 +153,26 @@ public function authWithToken(Request $request, int $id, string $token): Redirec
 
         return Redirect::route('person.show', ['id' => $id]);
     }
+
+    public function setPin(Request $request): JsonResponse
+    {
+        $id = $request->input('id') ?? null;
+        $token = $request->input('token') ?? null;
+        $pin = $request->input('pin') ?? null;
+
+        if ($id === null || $token === null || $pin === null) {
+            return response()->json(['status' => 'error', 'message' => 'invalid parameters'], 400);
+        }
+
+        $person = Person::findOrFail($id);
+
+        if ($person->auth_token != $token) {
+            return response()->json(['status' => 'error', 'message' => 'no permission'], 403);
+        }
```

Comment:
```suggestion
        $person = Person::findOrFail($id);
        if ($this->validateAuthToken($request, $person)) {
            return $this->validateAuthToken($request, $person, false);
        }
```

Maybe we should use this as used in other functions in this file?

Reply:
i dont know if i want to use it. 
When using the function validateAuthToken we would redirect to the error page. 
Using axios i can just clear the pin and stay on the same page.

Just to make sure when somehow a bug occours we stay on the same page on the tablet. When we redirect to the error page we would have to reset the tablet kiosk app, because there is no return button on that page.

What do you think? If you say it is a no issue i can change it.

Reply:
Should work in the newest commit